server.go

```diff
@@ -776,17 +813,20 @@ func shouldDropLocalConnection(local, remote *btcec.PublicKey) bool {
 	return bytes.Compare(localPubBytes, remotePubPbytes) > 0
 }
 
-// inboundPeerConnected initializes a new peer in response to a new inbound
+// InboundPeerConnected initializes a new peer in response to a new inbound
 // connection.
-func (s *server) inboundPeerConnected(conn net.Conn) {
-	s.peersMtx.Lock()
-	defer s.peersMtx.Unlock()
+//
+// NOTE: This function is safe for concurrent access.
+func (s *server) InboundPeerConnected(conn net.Conn) {
+	nodePub := conn.(*brontide.Conn).RemotePub()
+	pubStr := string(nodePub.SerializeCompressed())
 
 	// If we already have an inbound connection to this peer, then ignore
 	// this new connection.
-	nodePub := conn.(*brontide.Conn).RemotePub()
-	pubStr := string(nodePub.SerializeCompressed())
+	s.mu.Lock()
```

Comment:
Originally this was done to keep consistency with the OutboundPeerConnected, but I think it makes sense now to do as you suggest, which is overall simpler and easier to understand :)